net/libfab/libfab.c

```diff
@@ -508,6 +522,50 @@ static uint32_t libfab_handle_connect_request_events(struct m0_fab__tm *tm)
 	return 0;
 }
 
+/**
+ * Check connetion and shutdown events for tx ep 
+ */
+static void libfab_txep_event_check(struct m0_fab__ep *txep,
+				    struct m0_fab__active_ep *aep,
+				    struct m0_fab__tm *tm)
+{
+	int rc;
+
+	rc = libfab_check_for_event(aep->aep_tx_res.ftr_eq);
+	if (rc == FI_CONNECTED) {
+		libfab_pending_bufs_send(txep);
+		aep->aep_tx_state = FAB_CONNECTED;
+	} else if (rc == FI_SHUTDOWN) {
+		/* Reset and reopen endpoint */
+		libfab_txep_init(aep, tm);
+	}
+}
+
+/**
+ * Check for completion events on the for the rx ep
+ */
+static void libfab_rxep_comp_read(struct fid_cq *cq, struct m0_fab__ep *ep)
+{
+	struct m0_fab__buf *buf = NULL;
+	m0_bindex_t         len = 0;
+	int                 rc;
+
+	if (cq != NULL) {
+		rc = libfab_check_for_comp(cq, &buf, &len);
+		if (buf != NULL) {
```

Comment:
For a function call, we usually check the return value first, and then check output parameters.

Reply:
Done